goalie/mesh_seq.py

```diff
@@ -342,28 +341,41 @@ def _outputs_consistent(self):
         Assert that function spaces, initial conditions, and forms are given in a
         dictionary format with :attr:`MeshSeq.fields` as keys.
         """
-        for method in ["get_function_spaces", "get_initial_condition", "get_form"]:
-            if getattr(self, f"_{method}") is None:
+        for method in ["function_spaces", "initial_condition", "form", "solver"]:
+            if getattr(self, f"_get_{method}") is None:
                 continue
-            method_map = getattr(self, method)
-            if method == "get_function_spaces":
+            method_map = getattr(self, f"get_{method}")
+            if method == "function_spaces":
                 method_map = method_map(self.meshes[0])
-            elif method == "get_initial_condition":
+            elif method == "initial_condition":
                 method_map = method_map()
-            elif method == "get_form":
-                solution_map = {}
-                u = {}
-                for f in self.fields:
-                    u[f] = firedrake.Function(self.function_spaces[f][0])
-                    solution_map[f] = (u[f], u[f])
-                method_map = method_map()(0, solution_map)
-            assert isinstance(method_map, dict), f"{method} should return a dict"
+            elif method == "form":
+                self._reinitialise_fields(self.get_initial_condition())
+                method_map = method_map()(0)
+            elif method == "solver":
+                self._reinitialise_fields(self.get_initial_condition())
+                solver_gen = method_map()(0)
+                assert hasattr(solver_gen, "__next__"), "solver should yield"
+                next(solver_gen)
+                try:
+                    f, f_ = self.fields[next(iter(self.fields.keys()))]
+                    if (f.dat.data != f_.dat.data).all():
+                        self.warning(
+                            "solution unchanged after first timestep."
+                            " Does the solver yield before updating lagged solutions?"
```

Comment:
I think these kind of queries should only be done in debugging mode.

Also, minor point: capitalise first character of warning message for consistency with other warnings and errors.

Reply:
I realise now that `"solution unchanged after first timestep."` is also a bad description... Will fix

Reply:
Actually I agree - sorry for complicating. Done as you suggested in 2737ad669b93f9a340b2799aa5351e08ceb36a65